lib/fhir_models/fhir_ext/structure_definition.rb

```diff
@@ -117,12 +117,11 @@ def is_compatible?(another_definition)
           compare_extension_definition(x, y, another_definition)
         end
         y = get_extension(x.type[0].profile, right_extensions)
-        if !y.nil? && x.name != y.name
-          # both profiles share the same extension definition but with a different name
-          checked_extensions << x.name
-          checked_extensions << y.name
-          compare_element_definitions(x, y, another_definition)
-        end
+        next unless !y.nil? && x.name != y.name
```

Comment:
In #21, @wesrich suggested we change `.nil?` calls to just checking for truthiness (e.g. `y` instead of `!y.nil?`); I assume we want to do the same where these occur here?